gallery_dl/extractor/boosty.py

```diff
@@ -219,6 +219,107 @@ def items(self):
             yield Message.Queue, url, user
 
 
+class BoostyDirectMessagesExtractor(BoostyExtractor):
+    """Extractor for boosty.to direct messages"""
+    subcategory = "direct-messages"
+    directory_fmt = ("{category}", "{user[blogUrl]} ({user[id]})",
+                     "direct-messages")
+    pattern = BASE_PATTERN + r"/app/messages\?dialogId=(\d+)"
+    example = "https://boosty.to/app/messages?dialogId=123456"
+
+    def items(self):
+        """Yield direct messages from a given dialog ID."""
+        dialog_id = self.groups[0]
+        response = self.api.dialog(dialog_id)
+        sign = response.get("signedQuery", "")
+        messages = response.get("messages", {}).get("data", [])
+        chatmate = response.get("chatmate", {})
+        user = self.api.user(chatmate.get("url", ""))
+
+        for message in messages:
+            files = self._process_dialog(message, sign=sign)
+
+            if files:
+                data = {
+                    "post": message,
+                    "user": user,
+                    "count": len(files),
+                }
+                yield Message.Directory, data
+
+                for data["num"], file in enumerate(files, 1):
+                    data["file"] = file
+                    url = file["url"]
+                    yield Message.Url, url, text.nameext_from_url(url, data)
+
+        for message in self.api.dialog_messages(dialog_id=dialog_id,
+                                                offset=messages[0]["id"]):
+            files = self._process_dialog(message, sign=sign)
+            if files:
+                data = {
+                    "post": message,
+                    "user": user,
+                    "count": len(files),
+                }
+                yield Message.Directory, data
+
+                for data["num"], file in enumerate(files, 1):
+                    data["file"] = file
+                    url = file["url"]
+                    yield Message.Url, url, text.nameext_from_url(url, data)
+
+    def _process_dialog(self, post, sign=None):
+        """Process a post and return the list of files."""
+        files = []
+
+        for block in post.get("data", []):
```

Comment:
This looks a lot like the code in `_process_post`
https://github.com/mikf/gallery-dl/blob/182b54421709c29dfb9839393db3c2229d0690c0/gallery_dl/extractor/boosty.py#L82

Could you combine the block extraction code into a separate function to not have it duplicated here?